include/types.h

```diff
@@ -54,3 +54,12 @@ MAKE_PHANTOM_TYPE(node_id, size_t)
 MAKE_PHANTOM_TYPE(command_id, size_t)
 MAKE_PHANTOM_TYPE(collective_group_id, size_t)
 MAKE_PHANTOM_TYPE(reduction_id, size_t)
+MAKE_PHANTOM_TYPE(host_object_id, size_t)
+
+
+// declared in this header for include-dependency reasons
+namespace celerity::experimental {
+
+enum class side_effect_order { sequential };
```

Comment:
Bikeshedding: should `sequential` be all lower case here? I think in rest of the code base we use SCREAM_CASE for these. Do we want to change that in the long run?

Reply:
SYCL enums are `snake_case`, as are all SYCL-like enums we introduce ourselves (see `sycl_wrappers.h`). SCREAM_CASE is only used internally for enums so far, and AFAIC should be reserved for preprocessor defines (semantics: IDENTIFIER can expand to arbitrary token lists).